lib/setup/profile.js

```diff
@@ -66,6 +66,110 @@ var appendScript = function(name, options) {
+ * 3. If both exist, but neither sources the other,
+ *    append to .bashrc if platform is 'linux',
+ *    or append to .bash_profile if platform is 'darwin' (i.e. macOS)
+ *
+ * @private
+ * @function setup.profile.~shellInitScriptChooser
+ * @return {Promise}
+ */
+var shellInitScriptChooser = function() {
+  var bashrcFile = path.join(process.env.HOME, '.bashrc');
+  var bashProfileFile = path.join(process.env.HOME, '.bash_profile');
+  var bashrcExists;
+  var bashProfileExists;
+
+  // Check if .bashrc exists
+  var bashrcPromise = new Promise(function(resolve) {
+    fs.access(bashrcFile, function(err) {
+      bashrcExists = !err;
+      resolve();
+    });
+  });
+
+  // Check if .bash_profile exists
+  var bashProfilePromise = new Promise(function(resolve) {
+    fs.access(bashProfileFile, function(err) {
+      bashProfileExists = !err;
+      resolve();
+    });
+  });
+
+  var promise = Promise.all([bashrcPromise, bashProfilePromise])
+  .then(function() {
+    if ((bashrcExists || bashProfileExists) &&
+       !(bashrcExists && bashProfileExists)) { // ^ == bitwise XOR
+      return (bashrcExists && bashrcFile) ||
+             (bashProfileExists && bashProfileFile);
+    }
+
+    if (bashrcExists && bashProfileExists) {
+      var innerPromise = new Promise(function(resolve) {
+        fs.readFile(bashProfileFile, 'utf8', function(err, data) {
+          // RegExp to detect "~/.bashrc",
+          // "$HOME/.bashrc", or "/home/user/.bashrc"
+          var bashrcRe = new RegExp(' ?source +(\\\\\\n)? *(~\/.bashrc|' +
+            '\"?\\$HOME\/.bashrc\"?|[\'\"]?\/home\/\\w+\/.bashrc[\'\"]?)');
+
+          if (bashrcRe.test(data)) {
+            resolve(bashrcFile);
+          }
+          else {
+            resolve();
+          }
+        });
+      })
+      .then(function(appendFile) {
+        if (appendFile) {
+          return appendFile;
+        }
+
+        var innerInnerPromise = new Promise(function(resolve) {
+          fs.readFile(bashrcFile, 'utf8', function(err, data) {
+            // RegExp to detect "~/.bash_profile",
+            // "$HOME/.bash_profile", or "/home/user/.bash_profile"
+            var bashProfileRe = new RegExp(' ?source +(\\\\\\n)? *' +
+              '(~\/.bash_profile|\"?\\$HOME\/.bash_profile\"?|' +
+              '[\'\"]?\/home\/\\w+\/.bash_profile[\'\"]?)');
+
+            if (bashProfileRe.test(data)) {
+              resolve(bashProfileFile);
+            }
+            else {
+              resolve();
+            }
+          });
+        });
+
+        return innerInnerPromise;
+      })
+      .then(function(appendFile) {
+        if (appendFile) {
+          return appendFile;
+        }
+
+        // Neither .bashrc nor .bash_profile sources the other
+        var platformsToInitScript = {
+          linux:  bashrcFile,
+          darwin: bashProfileFile,
+        };
+
+        return platformsToInitScript[process.platform];
```

Comment:
There should probably be a fallback here in case the platform is not in this list.